core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigurationException.java

```diff
@@ -0,0 +1,46 @@
+package io.quarkus.runtime.configuration;
+
+/**
+ * An exception indicating that a configuration failure has occurred.
+ */
+public class ConfigurationException extends RuntimeException {
```

Comment:
We already have `ConfigurationError`. How is this one positioned?

Reply:
We need it to be in the runtime module because the run time can throw this exception at start.  The other is in the deployment module and probably should be replaced with this one at some point (this can be a follow-up item).

Reply:
#5704